src/Build/Construction/ProjectRootElement.cs

```diff
@@ -1887,6 +1895,55 @@ internal void MarkAsExplicitlyLoaded()
         }
 
         /// <summary>
+        /// Creates and returns a list of <see cref="ProjectImportElement"/> nodes which are implicitly
+        /// referenced by the Project.
+        /// </summary>
+        /// <param name="currentProjectOrImport">Current project</param>
+        /// <returns>An <see cref="IEnumerable{ReferencedSdk}"/> containing details of the SDKs referenced by the project.</returns>
+        internal List<ProjectImportElement> GetImplicitImportNodes(ProjectRootElement currentProjectOrImport)
+        {
+            var nodes = new List<ProjectImportElement>();
+
+            foreach (var referencedSdk in ParseSdks(Sdk, SdkLocation))
+            {
+                nodes.Add(ProjectImportElement.CreateImplicit("Sdk.props", currentProjectOrImport, ImplicitImportLocation.Top, referencedSdk));
+                nodes.Add(ProjectImportElement.CreateImplicit("Sdk.targets", currentProjectOrImport, ImplicitImportLocation.Bottom, referencedSdk));
+            }
+
+            foreach (var sdkNode in Children.OfType<ProjectSdkElement>())
```

Comment:
What if, the `ProjectSdkElement` has a condition on it? The ProjectElementContainer allows for it. Then, imports that are placed above an Sdk element might affect that condition via conditions / file system side effects.